cli/pkg/local/app.go

```diff
@@ -182,9 +184,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		"rill.download_row_limit": "0", // 0 means unlimited
 		"rill.stage_changes":      "false",
 	}
-	for k, v := range opts.Variables {
-		vars[k] = v
+	maps.Copy(vars, opts.Variables)
+	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, err
 	}
+	maps.Copy(vars, envMap)
```

Comment:
**Security Issue**: The use of 'godotenv.Read' to load environment variables from a '.env' file without proper error handling and security checks can lead to sensitive information leakage or unauthorized access if the environment variables contain sensitive data. <br> **Fix**: Implement additional checks to ensure that the '.env' file does not contain sensitive information that should not be loaded into the application context. Additionally, restrict access to the '.env' file to prevent unauthorized reading or modification. <br> **Code Suggestion**: 
 ```
 + if err != nil and !errors.Is(err, os.ErrNotExist) {
 +     log.Error("Failed to read .env file", err)
 +     return nil, err
 + }
 + if !validateEnvMap(envMap) {
 +     log.Error("Invalid .env file content")
 +     return nil, fmt.Errorf("invalid .env file content")
 + }
 + maps.Copy(vars, envMap)
 ```